### ISSUE TITLE
✅ 🔗 📚 🐞 Bug Report: Query Control Fallback Query Bug  (Duplicate of #21436) - [ED-10701]

### ISSUE THREAD

**Issue by amyriounis** [elementor/elementor#23259]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

In `/elementor-pro/modules/query-control/classes/elementor-related-query.php` an optimization is being applied that essentially disables the whole fallback query concept.

In more detail, in the `get_fallback_query` function there is this line of code:
```php
$this->fallback_args['no_found_rows'] = true;
``` 

which disables the standard row-counting behavior on the returned query. This will result in the `$query->found_posts` value to always being set to `0`.

The issue is that this value is being used on most query-related widgets to check for post existence. For example, in
 `/elementor-pro/modules/posts/skins/skin-base.php`, inside the `render()` function there is this check that will **always** fail for fallback queries:
 ```php
if ( ! $query->found_posts ) {
	return;
}
```

This will result in the widget aborting rendering altogether.

### Steps to reproduce

Let's use a Related Posts example to illustrate the issue:

1. Create a post with a unique category.
2. Create a basic single post template.
3. Add a posts widget to the template, to be used as a Related Posts section.
4. In the Query section, set Source to "Related",  Include By to "Term", Term to "Categories" and Fallback to "Recent Posts", like so:
![rewrwer](https://github.com/elementor/elementor/assets/9055437/76b78bef-9cfb-4f08-960d-9b37cab32e0d)
5. Save the template.
6. In the frontend, navigate to the post you created in step 1. 

Normally, because the post has no related posts (posts in the same category) the widget should have displayed the most recent posts. The reality is that the widget does not render at all due to the `$query->found_posts` check failing every time.

### Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt


					
== Server Environment ==
	Operating System: Linux
	Software: nginx/1.24.0
	MySQL version: Ubuntu 18.04 v5.6.49-89.0
	PHP Version: 8.1.18
	PHP Memory Limit: 256M
	PHP Max Input Vars: 20000
	PHP Max Post Size: 100M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2.2
	Site URL: localhost
	Home URL: localhost
	WP Multisite: No
	Max Upload Size: 100 MB
	Memory limit: 128M
	Max Memory limit: 256M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.8.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.82

== Active Plugins ==
	Elementor
		Version: 3.15.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.15.0
		Author: Elementor.com


== Elements Usage ==
	
	archive : 2
		icon-list : 2
		loop-grid : 2
		template : 1
		text-editor : 1
		theme-archive-title : 2
	footer : 2
		button : 4
		divider : 1
		form : 1
		heading : 10
		icon : 3
		icon-list : 1
		image : 4
		nav-menu : 6
		social-icons : 2
		spacer : 2
		text-editor : 3
		theme-site-title : 1
	header : 3
		button : 6
		icon : 4
		image : 4
		nav-menu : 4
	page : 21
		blockquote : 1
		button : 8
		call-to-action : 2
		divider : 4
		heading : 34
		icon : 2
		icon-box : 25
		icon-list : 11
		image : 4
		loop-carousel : 4
		menu-anchor : 3
		social-icons : 1
		spacer : 3
		template : 1
		text-editor : 34
	popup : 8
		button : 21
		call-to-action : 12
		heading : 11
		image : 2
		loop-grid : 3
		search-form : 1
		spacer : 4
		text-editor : 2
	section : 6
		divider : 1
		heading : 2
		icon-list : 2
		post-comments : 1
		spacer : 3
		text-editor : 2
	widget : 0
		button : 1
		icon-list : 1
	wp-post : 42
		blockquote : 7
		button : 12
		call-to-action : 9
		heading : 60
		icon-box : 6
		icon-list : 19
		image : 7
		loop-carousel : 1
		template : 1
		text-editor : 59
	wp-page : 52
		animated-headline : 1
		button : 108
		call-to-action : 18
		divider : 36
		form : 2
		heading : 225
		icon : 32
		icon-box : 174
		icon-list : 55
		image : 61
		loop-carousel : 27
		loop-grid : 4
		menu-anchor : 9
		nav-menu : 2
		posts : 1
		social-icons : 8
		spacer : 21
		star-rating : 1
		template : 9
		text-editor : 174
		toggle : 1
	single-post : 5
		button : 3
		call-to-action : 1
		divider : 4
		form : 1
		heading : 5
		icon-list : 8
		image : 1
		loop-carousel : 2
		loop-grid : 1
		menu-anchor : 1
		nav-menu : 2
		post-info : 3
		spacer : 4
		text-editor : 4
		theme-post-content : 3
		theme-post-title : 3
		toggle : 1
	single-page : 5
		button : 3
		heading : 2
		icon : 1
		icon-list : 3
		image : 1
		template : 2
		text-editor : 2
		theme-post-content : 4
		theme-post-excerpt : 1
		theme-post-title : 2
	search-results : 1
		archive-posts : 1
		heading : 1
		search-form : 1
		spacer : 1
		text-editor : 1
		theme-archive-title : 1
	loop-item : 14
		button : 11
		divider : 3
		heading : 1
		icon : 4
		icon-box : 3
		image : 1
		post-info : 11
		spacer : 2
		theme-post-content : 1
		theme-post-excerpt : 4
		theme-post-featured-image : 6
		theme-post-title : 1
	error-404 : 1
		counter : 1
		heading : 1
		icon : 1
		search-form : 1
		text-editor : 1


== Settings ==
	
	cpt_support: post, page
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap
	load_fa4_shim: yes


== Features ==
	Custom Fonts: 1
	Custom Icons: 1

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active
	Improved Asset Loading: Active
	Improved CSS Loading: Active
	Inline Font Icons: Inactive
	Additional Custom Breakpoints: Inactive
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Inactive
	Upgrade Swiper Library: Active
	Grid Container: Inactive
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor Top Bar: Inactive
	Landing Pages: Active by default
	Nested Elements: Inactive
	Lazy Load Background Images: Inactive
	Global Style Guide: Inactive
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Inactive
	Taxonomy Filter: Inactive by default


== Log ==
	
== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	

				

````

</details>

**Comment by jfama-purecobalt** [comment]
Hey there @amyriounis 
This is effectively already reported here: https://github.com/elementor/elementor/issues/21436. This was reported back in February by [trinermedia](https://github.com/trinermedia) and has been confirmed as an reproducible issue by several users. Not sure why the Elementor team is dragging their heels on this one.

**Comment by nicholaszein** [comment]
Hello @amyriounis!

Thank you for your submission! 🙏 

We are aware of this issue and our team is already working on a fix. We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding! 😉

### 🔗 🐞 This issue is a Duplicate of #21436 

Which is on the same subject. 

For this reason, I'm closing this thread and 🔃 **Merging** it with the original, so we can have all the discussions in one location, and deal with it more efficiently.

Feel free to continue the conversation there. 😉 

### 🔍 Please, in the future, remember to check if there isn't any thread similar to yours already opened. 

Kind regards

**Comment by nicholaszein** [comment]
Hey @amyriounis!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.16! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
